src/hotspot/share/runtime/globals.hpp

```diff
@@ -295,6 +295,9 @@ const int ObjectAlignmentInBytes = 8;
   product(bool, UseInlineCaches, true,                                      \
           "Use Inline Caches for virtual calls ")                           \
                                                                             \
+  product(uintx, InlineCacheBufferSize, 10*K, EXPERIMENTAL,                 \
+          "InlineCacheBuffer size")                                         \
```

Comment:
Can you make this type an int and add a range to it?  Line 143 above will get -Wconversion warnings if we ever turned them on.

Reply:
Thanks, I changed type to int and added a range check constraint.

Reply:
I'd rather have the type as size_t and change StubQueue accordingly.

Reply:
Thanks @dean-long. 
I would like to keep this enhancement simple and minimal so that it can be backported to 17 and 11.
So I'd like to avoid changes to StubQueue. I can change the type of InlineCacheBufferSize to size_t and add checked_cast to StubQueue constructor in InlineCacheBuffer::initialize():
_buffer = new StubQueue(new ICStubInterface, checked_cast<int>(InlineCacheBufferSize), InlineCacheBuffer_lock, "InlineCacheBuffer"); 

Because in any case InlineCacheBufferSize can't be greater than INT_MAX:
`InlineCacheBufferSize < NonNMethodCodeHeapSize < ReservedCodeCacheSize < CODE_CACHE_DEFAULT_LIMIT = 2G`:
https://github.com/openjdk/jdk/blob/master/src/hotspot/share/code/codeCache.cpp#L191
https://github.com/openjdk/jdk/blob/master/src/hotspot/share/compiler/compilerDefinitions.cpp#L492
https://github.com/openjdk/jdk/blob/master/src/hotspot/share/utilities/globalDefinitions.hpp#L589

Will that be OK?

Reply:
OK.

Reply:
done